### PR TITLE
[cxx-interop] CxxShim files became arch-independent

### PR DESCRIPTION
**Pull request by egorzhdan**
This adjusts the Windows installation logic after the changes in https://github.com/apple/swift/pull/66765

rdar://110788977

### CODE REVIEW

File: platforms/Windows/sdk.wxs [hunk 1]
```diff
@@ -417,11 +417,11 @@
 
     <ComponentGroup Id="libcxxshim">
       <Component Directory="WindowsSDK_usr_lib_swift_windows_ARCH" Id="libcxxshim.h">
```
Comment (compnerd) [inline platforms/Windows/sdk.wxs:419]: This needs to change as well, as while the point where it is installed to was updated, the `Directory` attribute here controls where it is deployed to.

```suggestion
      <Component Directory="WindowsSDK_usr_lib_swift_windows" Id="libcxxshim.h">
```

**Review by compnerd — CHANGES_REQUESTED** [review]
Thanks!

File: platforms/Windows/sdk.wxs (hunk 1 above)
Comment (egorzhdan) [reply platforms/Windows/sdk.wxs]: Fixed, thanks!

**Review by compnerd — APPROVED** [review]
Probably a good idea to run a build.

**Pull request merged**